mobile/src/screens/BookScreen.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Text, View, StyleSheet, Image, ScrollView } from 'react-native';
+import { useRoute } from '@react-navigation/native';
+import { MarkdownView } from 'react-native-markdown-view';
+import YoutubePlayer from 'react-native-youtube-iframe';
+import { BookScreenRouteProps } from '../navigation/HomeStackNavigator';
+import { LoadingCircle } from '../components/LoadingCircle';
+import { APIContext } from '../context/APIContext';
+import { I18nContext } from '../context/I18nContext';
+import { TextStyles } from '../styles/TextStyles';
+import { ButtonGroup } from '../components/ButtonGroup';
+import { LanguageButtons } from '../components/LanguageButtons';
+
+/**
+ * Individual book view displaying book details
+ */
+export const BookScreen: React.FC = () => {
+  // get book id and langs from route params
+  const route = useRoute<BookScreenRouteProps>();
+  const { id, langs } = route.params;
+
+  const client = useContext(APIContext);
+  const i18nCtx = useContext(I18nContext);
+
+  // book screen states
+  const [loading, setLoading] = useState(true);
+  const [language, setLanguage] = useState(i18nCtx.locale.substring(0, 2));
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [image, setImage] = useState('');
+  const [tabBody, setTabBody] = useState('');
+  const [tabVideo, setTabVideo] = useState('');
+  const [activeButton, setActiveButton] = useState('');
+  const [tabContent, setTabContent] = useState(null);
+
+  const markdownStyles = {
+    heading: { fontFamily: 'Gotham-Light' },
+    paragraph: { fontFamily: 'Gotham-Light' },
+    strong: { fontFamily: 'Gotham-Bold' },
+    listItemNumber: { fontFamily: 'Gotham-Light' },
+    listItemOrderedContent: { fontFamily: 'Gotham-Light' },
+    listItemUnorderedContent: { fontFamily: 'Gotham-Light' },
+    tableHeaderCellContent: { fontFamily: 'Gotham-Light' },
+    em: { fontFamily: 'Gotham-Italic' },
+  };
+
+  // fetches book details on language change
+  useEffect(
+    () => {
+      setLoading(true);
+      let lang;
+
+      if (langs.indexOf(language) > -1) {
+        lang = language;
+      } else if (langs.indexOf('en') > -1) {
+        lang = 'en';
+        setLanguage(lang);
+      } else {
+        const [first] = langs;
+        lang = first;
+        setLanguage(lang);
+      }
```

Comment:
Ah here it is, yeah I think you can just determine this before the call to useState().